### PR TITLE
Restructuring

### PR DESCRIPTION
**Pull request by MarvinTeichmann**
Most of the training code is moved from tv-train to tensorvision/train.py. Additionally most of the methods get refractured. Especially the function run_training is now split up into several smaller functions. This will be useful when implementing tv_analysis.

### CODE REVIEW

**Comment by MartinThoma** [comment]
You should include "fixes #31" in one of the commit messages.

File: bin/tv-train [hunk 1]
```diff
@@ -8,15 +8,17 @@ from __future__ import division
 from __future__ import print_function
 
 import imp
+import pdb
```
Comment (MartinThoma) [inline bin/tv-train]: As far as I know this is for debugging, right? If so, please don't commit this.

File: bin/tv-train [hunk 2]
```diff
@@ -8,15 +8,17 @@ from __future__ import division
...
 import logging
+import numpy as np
 import os.path
 import sys
 
 # configure logging
 if 'TV_IS_DEV' in os.environ and os.environ['TV_IS_DEV']:
     logging.basicConfig(format='%(asctime)s %(levelname)s %(message)s',
-                        level=logging.DEBUG,
+                        level=logging.INFO,
```
Comment (MartinThoma) [inline bin/tv-train:22]: No! When you're in development mode, it should show the debug messages!

File: bin/tv-train [hunk 3]
```diff
@@ -136,14 +138,14 @@ def write_precision_to_summary(precision, summary_writer, name, global_step,
...
-def do_eval(hypes, eval_correct, phase, sess):
+def do_eval(hypes, eval_list, phase, sess):
     """Run one evaluation against the full epoch of data.
 
     Parameters
     ----------
     hypes : dict
         Hyperparameters
-    eval_correct : TODO
+    eval_dict : TODO
```
Comment (MartinThoma) [inline bin/tv-train]: This is not what you called it in the function signature

File: bin/tv-train (hunk 2 above)
Comment (MarvinTeichmann) [inline bin/tv-train:22]: Well, kind of. Write now using logging.DEBUG yields to really annoying Tensorflow debug output.

**Comment by MarvinTeichmann** [comment]
Merged in 1778343aa50ce547753e0f6e950ea11c07165d04

**Pull request closed without merging**